lib/internal/url.js

```diff
@@ -817,7 +817,8 @@ class URLSearchParams {
   constructor(init = undefined) {
     if (init === null || init === undefined) {
       this[searchParams] = [];
-    } else if (typeof init === 'object') {
+    } else if (typeof init === 'object' && init !== null ||
+               typeof init === 'function') {
```

Comment:
Not sure how others feel, but I prefer to use parentheses when I’m mixing logical and and logical or because I can never remember the precedence rules. ;)